jetbrains-ultimate/src-201/software/aws/toolkits/jetbrains/core/datagrip/auth/SecretsManagerAuth.kt

```diff
@@ -0,0 +1,78 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.core.datagrip.auth
+
+import com.fasterxml.jackson.databind.DeserializationFeature
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
+import com.intellij.credentialStore.Credentials
+import com.intellij.database.access.DatabaseCredentials
+import com.intellij.database.dataSource.DatabaseAuthProvider
+import com.intellij.database.dataSource.DatabaseAuthProvider.AuthWidget
+import com.intellij.database.dataSource.DatabaseConnectionInterceptor.ProtoConnection
+import com.intellij.database.dataSource.DatabaseCredentialsAuthProvider
+import com.intellij.database.dataSource.LocalDataSource
+import com.intellij.openapi.project.Project
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.future.future
+import software.amazon.awssdk.services.secretsmanager.SecretsManagerClient
+import software.aws.toolkits.core.utils.getLogger
+import software.aws.toolkits.core.utils.info
+import software.aws.toolkits.jetbrains.core.awsClient
+import software.aws.toolkits.jetbrains.core.credentials.ConnectionSettings
+import software.aws.toolkits.jetbrains.core.datagrip.getAwsConnectionSettings
+import software.aws.toolkits.jetbrains.utils.ApplicationThreadPoolScope
+import software.aws.toolkits.resources.message
+import java.util.concurrent.CompletionStage
+
+data class SecretsManagerConfiguration(
+    val connectionSettings: ConnectionSettings,
+    val secretId: String
+)
+
+class SecretsManagerAuth : DatabaseAuthProvider, CoroutineScope by ApplicationThreadPoolScope("RedshiftSecretsManagerAuth") {
+    private val objectMapper = jacksonObjectMapper().disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES)
+
+    override fun getId(): String = providerId
+    override fun isApplicable(dataSource: LocalDataSource): Boolean = dataSource.dbms.isRedshift || dataSource.dbms.isMysql || dataSource.dbms.isPostgres
+    override fun getDisplayName(): String = message("datagrip.auth.secrets_manager")
+
+    override fun createWidget(creds: DatabaseCredentials, source: LocalDataSource): AuthWidget? = SecretsManagerAuthWidget()
+
+    override fun intercept(
+        connection: ProtoConnection,
+        silent: Boolean
+    ): CompletionStage<ProtoConnection>? {
+        LOG.info { "Intercepting db connection [$connection]" }
+        return future {
```

Comment:
Should match the pattern with the other PR

Reply:
This is how the other two look, what do you mean?